src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -161,6 +162,36 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
         return result;
     }
 
+    /**
+     * Parse block comment DetailAST as Javadoc DetailNode tree.
+     * @param blockComment DetailAST
+     * @return DetailNode tree
+     */
+    public DetailNode parseJavadocAsDetailNode(DetailAST blockComment) {
+        final ParseStatus status = parseJavadoc(blockComment);
+        if (status.getParseErrorMessage() != null) {
+            throw new IllegalArgumentException(getParseErrorMessage(status.getParseErrorMessage()));
+        }
```

Comment:
This should be the CheckstyleException. I decided to leave it as it is, maybe there's a reason.